### PR TITLE
Add clinic

### PR DESCRIPTION
**Pull request by Elexy**
Tests are now defined at startup based on STAGE=local or not since that doesn't change at runtime.
Each individual test now has definitions for several clinic runs. Each type of clinic tool is run with a predetermined `fileSet`.

In this PR the output (/tmp/out) is cleaned up after each test is complete. This assumes that the test runner itself will upload the clinic results somewhere.
```
{ name: 'localTransfer',
    benchmark:
     'killall node 2>/dev/null; source ~/.nvm/nvm.sh &&  OUT_FOLDER=/tmp/out REMOTE=true  node ~/ipfs/tests//local-transfer.js',
    doctor:
     [ 'killall node 2>/dev/null; source ~/.nvm/nvm.sh &&  FILESET="One4MBFile" clinic doctor --dest /tmp/out/localTransfer/ -- node ~/ipfs/tests//local-transfer.js',
       'killall node 2>/dev/null; source ~/.nvm/nvm.sh &&  FILESET="One128MBFile" clinic doctor --dest /tmp/out/localTransfer/ -- node ~/ipfs/tests//local-transfer.js',
       'killall node 2>/dev/null; source ~/.nvm/nvm.sh &&  FILESET="OneGBFile" clinic doctor --dest /tmp/out/localTransfer/ -- node ~/ipfs/tests//local-transfer.js' ],
    flame:
     [ 'killall node 2>/dev/null; source ~/.nvm/nvm.sh &&  FILESET="One4MBFile" clinic flame --dest /tmp/out/localTransfer/ -- node ~/ipfs/tests//local-transfer.js',
       'killall node 2>/dev/null; source ~/.nvm/nvm.sh &&  FILESET="One128MBFile" clinic flame --dest /tmp/out/localTransfer/ -- node ~/ipfs/tests//local-transfer.js',
       'killall node 2>/dev/null; source ~/.nvm/nvm.sh &&  FILESET="OneGBFile" clinic flame --dest /tmp/out/localTransfer/ -- node ~/ipfs/tests//local-transfer.js' ],
    bubbleProf:
     [ 'killall node 2>/dev/null; source ~/.nvm/nvm.sh &&  FILESET="One4MBFile" clinic bubbleProf --dest /tmp/out/localTransfer/ -- node ~/ipfs/tests//local-transfer.js',
       'killall node 2>/dev/null; source ~/.nvm/nvm.sh &&  FILESET="One128MBFile" clinic bubbleProf --dest /tmp/out/localTransfer/ -- node ~/ipfs/tests//local-transfer.js' ] }
```

### CODE REVIEW

File: runner/config.js [hunk 1]
```diff
@@ -40,37 +45,84 @@ const getBenchmarkHostname = () => {
...
+
+const clinicRuns = {
+  doctor: {
+    fileSets: ['One4MBFile', 'One128MBFile', 'OneGBFile']
+  },
+  flame: {
+    fileSets: ['One4MBFile', 'One128MBFile', 'OneGBFile']
+  },
+  bubbleProf: {
+    fileSets: ['One4MBFile', 'One128MBFile']
```
Comment (mcollina) [inline runner/config.js]: One128MBFile would likely crash bubbleprof at this point :/

Comment (Elexy) [reply runner/config.js]: it didn't yet, but I'll use a smaller one to be safe

**Comment by Elexy** [comment]
This now assumes clinic runs are uploaded after output files are written. The runner retrieves the SHA from that run and outputs is to the console for now:
```
INFO [1544717667878] (38780 on mbpro-2277.local):
    benchmark: {
      "name": "unixFsAdd",
      "fileSet": "One4MBFile"
    }
    clinic: {
      "operation": "doctor",
      "sha": {
        "sha": "c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2"
      }
    }
    ipfs: {
      "commit": "tbd"
    }
```

File: tests/upload-ipfs.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+OUTPATH=$1
+TESTNAME=$2
+
+# Usage info
+usage() {
+cat << EOF
+
+Usage: ${0##*/} <ouputPath> <TestName>
+
+Upload a directory to ipfs and store the SHA in a json file at <ouput path>/<Test name>.json
```
Comment (mcollina) [inline tests/upload-ipfs.sh]: I think we should aim to do a single upload with:

1. all json files inside a triggered run
2. all clinic files of that run

So, one single SHA for all the runs.

Comment (Elexy) [reply tests/upload-ipfs.sh]: That sounds good. Requires a refactor to leave all the generated results on the benchmark machine until the run is over.
Structure could be mirror the test config:
```
localTransfer
- benchmark
-- results.json
- doctor
-- warmup
--- One4MBFile
---- <pid>.clinic-doctor.html
---- <pid>.clinic-doctor
--- One128MBFile
---- <pid>.clinic-doctor.html
---- <pid>.clinic-doctor
-- nowarmup
--- One4MBFile
---- <pid>.clinic-doctor.html
---- <pid>.clinic-doctor
--- One128MBFile
---- <pid>.clinic-doctor.html
---- <pid>.clinic-doctor
localAdd
- benchmark
-- results.json
- doctor
-- warmup
--- One4MBFile
---- <pid>.clinic-doctor.html
---- <pid>.clinic-doctor
--- One128MBFile
---- <pid>.clinic-doctor.html
---- <pid>.clinic-doctor
-- nowarmup
--- One4MBFile
---- <pid>.clinic-doctor.html
---- <pid>.clinic-doctor
--- One128MBFile
---- <pid>.clinic-doctor.html
---- <pid>.clinic-doctor
```
  Does that make sense @mcollina ?

Comment (mcollina) [reply tests/upload-ipfs.sh]: yes that's ok.

**Pull request merged**